src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -122,11 +122,11 @@ def collect_tool_methods_with_init_inputs_in_module(m):
 
 def _parse_tool_from_function(f, initialize_inputs=None, gen_custom_type_conn=False, skip_prompt_template=False):
     try:
-        tool_type = getattr(f, "__type") or ToolType.PYTHON
+        tool_type = getattr(f, "__type", None) or ToolType.PYTHON
     except Exception as e:
         raise e
-    tool_name = getattr(f, "__name")
-    description = getattr(f, "__description")
+    tool_name = getattr(f, "__name", None)
+    description = getattr(f, "__description", None)
```

Comment:
this could be None, and inspect.getdoc could be None, is it mandatory to give a default value for this?

Reply:
There is no need to set default value for this field.
This field should be set by the user himself, if not set, this field will also be skipped in the generated tool meta